Experimental/CameraComponent/Viewport.cs

```diff
@@ -0,0 +1,322 @@
+                    // wide
+                    if (size.X >= size.Y)
+                    {
+                        stretchX = false;
+                        matStretch = StretchMatrix(stretchX);
+                    }
+                    // long
+                    else
+                    {
+                        stretchX = true;
+                        matStretch = StretchMatrix(stretchX);
+                    }
+                    break;
+                case Stretch.UniformToFill:
+                    // wide
+                    if (size.X >= size.Y)
+                    {
+                        stretchX = true;
+                        matStretch = StretchMatrix(stretchX);
+                    }
+                    // long
+                    else
+                    {
+                        stretchX = false;
+                        matStretch = StretchMatrix(stretchX);
+                    }
+                    break;
+            }
+
+            return matStretch;
+        }
+
+        public void CreateExpressionAnimation()
+        {
+            propertySet = Window.Current.Compositor.CreatePropertySet();
+            propertySet.InsertVector3("Offset", offset);
+            propertySet.InsertVector2("Size", size);
+            propertySet.InsertScalar("Stretch", (float)stretch);
+            propertySet.InsertMatrix4x4("ViewMatrix", camera.CreateViewMatrix());
+            propertySet.InsertMatrix4x4("NormalizingMatrix", camera.Projection.CreateNormalizingMatrix());
+            propertySet.InsertMatrix4x4("StretchMatrix", this.CreateStretchMatrix());
+
+            camera.Projection.CreateExpressionAnimation(propertySet, "NormalizingMatrix");
+            camera.CreateExpressionAnimation(propertySet, "ViewMatrix");
+            propertySet.StartAnimation("StretchMatrix", this.CreateStretchExpressionMatrix());
+
+            var cameraMatExpression = Window.Current.Compositor.CreateExpressionAnimation();
+            cameraMatExpression.Expression = "Viewport.ViewMatrix * Viewport.NormalizingMatrix * Viewport.StretchMatrix " +
+                "* Matrix4x4.CreateTranslation(Vector3(Viewport.Offset.X + Viewport.Size.X / 2f, Viewport.Offset.Y + Viewport.Size.Y / 2f, Viewport.Offset.Z))";
+            cameraMatExpression.SetReferenceParameter("Viewport", propertySet);
+
+            sceneVisual.StartAnimation("TransformMatrix", cameraMatExpression);
+        }
+
+        private Matrix4x4 StretchMatrix(bool stretchX)
+        {
+            Matrix4x4 matScale = Matrix4x4.Identity;
+            matScale.M11 = matScale.M22 = matScale.M33 = stretchX ? size.X : size.Y;
+
+            return matScale;
+        }
+        private ExpressionAnimation CreateStretchExpressionMatrix()
+        {
+            string stretchMat;
+
+            float stretch;
+            propertySet.TryGetScalar("Stretch", out stretch);
+
+            switch (stretch)
+            {
+                case 0f: // Fill

```

Comment:
I recommend substituting your ternary expression in here.

Reply:
Resolved